### PR TITLE
Fix wrong looping over indices of OffsetArray

### LINKED ISSUE #2815: Dangerous behavior with `@inbounds` within for loop over `1:lenght(ξ::OffSetArray)`
This is a tricky point -- We should be very careful here. It can easily may be a bug!?

https://github.com/CliMA/Oceananigans.jl/blob/91341616c313bc48cf122945a2e138c9442e21de/src/Grids/conformal_cubed_sphere_face_grid.jl#L127-L129

`ξ` and `η` are offset arrays...!

Perhaps it's better to replace `1:length(ξ)` with `eachindex(ξ)`; see https://docs.julialang.org/en/v1/devdocs/boundscheck/#Caution!

### PR DESCRIPTION
**Pull request by navidcy**
~see [https://docs.julialang.org/en/v1/devdocs/boundscheck/\#Caution](https://docs.julialang.org/en/v1/devdocs/boundscheck/#Caution)~

Closes #2815

### CODE REVIEW

**Comment by glwagner** [comment]
The docs link is broken. What does it say?

File: src/Grids/conformal_cubed_sphere_face_grid.jl [hunk 1]
```diff
@@ -115,7 +115,7 @@ function ConformalCubedSphereFaceGrid(arch::AbstractArchitecture = CPU(),
     ZS = (Zᶜᶜᵃ, Zᶠᶜᵃ, Zᶜᶠᵃ, Zᶠᶠᵃ)
 
     for (ξ, η, X, Y, Z) in zip(ξS, ηS, XS, YS, ZS)
-        for i in 1:length(ξ), j in 1:length(η)
+        for i in 1:Nξ, j in 1:Nη
```
Comment (glwagner) [inline src/Grids/conformal_cubed_sphere_face_grid.jl:118]: What does this have to do with julia docs? I agree that its correct now however, and that previously it was wrong.

**Comment by navidcy** [comment]
> The docs link is broken. What does it say?

Fixed

File: src/Grids/conformal_cubed_sphere_face_grid.jl (hunk 1 above)
Comment (navidcy) [reply src/Grids/conformal_cubed_sphere_face_grid.jl:118]: It was just a note on how inbounds can be used wrong and give no error.

**Comment by glwagner** [comment]
Ah I see. That advice is about writing generic code, but we don't need that here.

File: src/Grids/conformal_cubed_sphere_face_grid.jl (hunk 1 above)
Comment (glwagner) [reply src/Grids/conformal_cubed_sphere_face_grid.jl:118]: It's not a problem with inbounds though right? It's a problem with the loop bounds being incorrect.

**Comment by navidcy** [comment]
Right.

File: src/Grids/conformal_cubed_sphere_face_grid.jl (hunk 1 above)
Comment (navidcy) [reply src/Grids/conformal_cubed_sphere_face_grid.jl:118]: Right

**Review by glwagner — APPROVED** [review]

**Pull request merged**